internal/storage/git/store.go

```diff
@@ -389,63 +387,66 @@ func (s *Store) updateIndex(ctx context.Context) error {
 
 	for _, c := range changes {
 		s.log.Debugw("Processing change", "change", c)
-		switch {
-		case c.From.Name == "" && s.inPolicyDir(c.To.Name): // File created
-			if filename := s.schemaFileName(c.To.Name); filename != "" {
-				s.NotifySubscribers(storage.NewSchemaEvent(storage.EventAddOrUpdateSchema, filename))
-				continue
-			}
 
-			if err := s.applyIndexUpdate(c.To, storage.EventAddOrUpdatePolicy); err != nil {
-				return err
-			}
-		case c.To.Name == "" && s.inPolicyDir(c.From.Name): // File deleted
-			if filename := s.schemaFileName(c.From.Name); filename != "" {
-				s.NotifySubscribers(storage.NewSchemaEvent(storage.EventDeleteSchema, filename))
-				continue
-			}
+		fromPath, fromType := s.normalizePath(c.From.Name)
+		toPath, toType := s.normalizePath(c.To.Name)
 
-			if err := s.applyIndexUpdate(c.From, storage.EventDeletePolicy); err != nil {
-				return err
-			}
-		case s.inPolicyDir(c.From.Name) && !s.inPolicyDir(c.To.Name): // File moved out of policy dir
-			if filename := s.schemaFileName(c.From.Name); filename != "" {
-				s.NotifySubscribers(storage.NewSchemaEvent(storage.EventDeleteSchema, filename))
-				continue
-			}
+		if fromPath != toPath || fromType != toType {
+			switch fromType {
+			case util.FileTypePolicy:
+				if err := s.applyIndexUpdate(c.From, storage.EventDeletePolicy); err != nil {
+					return err
+				}
 
-			if err := s.applyIndexUpdate(c.From, storage.EventDeletePolicy); err != nil {
-				return err
-			}
-		case s.inPolicyDir(c.To.Name): // file moved in or modified
-			if filename := s.schemaFileName(c.To.Name); filename != "" {
-				s.NotifySubscribers(storage.NewSchemaEvent(storage.EventAddOrUpdateSchema, filename))
-				continue
+			case util.FileTypeSchema:
+				s.NotifySubscribers(storage.NewSchemaEvent(storage.EventDeleteSchema, fromPath))
+
+			default:
+				s.log.Debugw("Not applying delete", "change", c)
 			}
+		}
 
+		switch toType {
+		case util.FileTypePolicy:
 			if err := s.applyIndexUpdate(c.To, storage.EventAddOrUpdatePolicy); err != nil {
 				return err
 			}
+
+		case util.FileTypeSchema:
+			s.NotifySubscribers(storage.NewSchemaEvent(storage.EventAddOrUpdateSchema, toPath))
+
 		default:
-			s.log.Debugw("Ignoring change", "change", c)
+			s.log.Debugw("Not applying add/update", "change", c)
 		}
 	}
 
 	s.log.Info("Index updated")
 	return nil
 }
 
-func (s *Store) applyIndexUpdate(ce object.ChangeEntry, eventKind storage.EventKind) error {
-	if !util.IsSupportedFileType(ce.Name) {
-		s.log.Infow("Ignoring unsupported file type", "path", ce.Name)
-		return nil
+func (s *Store) normalizePath(path string) (string, util.IndexedFileType) {
+	if path == "" {
+		return path, util.FileTypeNotIndexed
 	}
 
-	if util.IsSupportedTestFile(ce.Name) {
-		s.log.Infow("Ignoring test file", "path", ce.Name)
-		return nil
+	if s.conf.SubDir != "" {
+		relativePath := strings.TrimPrefix(path, strings.TrimSuffix(s.conf.SubDir, "/")+"/")
```

Comment:
Would this break on Windows where the separator is `\`? Shouldn't we call `filepath.ToSlash` first?

Reply:
Yeah, good point. `path` is sourced from `ChangeEntry.Name` which [is slash-separated](https://pkg.go.dev/github.com/go-git/go-git/v5/plumbing/object?utm_source=gopls#ChangeEntry.Name), but `s.conf.SubDir` could conceivably be backslash-separated.